biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/FastaStreamer.java

```diff
@@ -0,0 +1,179 @@
+		return this;
+	}
+
+	public FastaStreamer batchSize(int size) {
+		this.batchSize = size;
+		return this;
+	}
+
+	/**
+	 * Enable iteration through the proteins in the file using syntax such as:
+	 * <pre>
+	 *     for(ProteinSequence sequence : FastaStreamer.from(path).each()) {
+	 *         .
+	 *         .
+	 *         .
+	 *     }
+	 * </pre>
+	 *
+	 * @return an iterable suitable for an iteration loop
+	 */
+	public Iterable<ProteinSequence> each() {
+		return () -> stream().iterator();
+	}
+
+	/**
+	 * Create a stream of protein sequences from the contents of the path
+	 * @return the stream
+	 */
+	public Stream<ProteinSequence> stream() {
+		InputStreamProvider provider = new InputStreamProvider();
+		InputStream input;
+		try {
+			input = provider.getInputStream(getPath().toFile());
+		} catch (IOException exception) {
+			throw new RuntimeException(exception);
+		}
+		FastaReader<ProteinSequence, AminoAcidCompound> reader = new FastaReader<>(input, getHeaderParser(), getSequenceCreator());
+		Spliterator<ProteinSequence> source = new Spliterators.AbstractSpliterator<>(Integer.MAX_VALUE, Spliterator.IMMUTABLE | Spliterator.NONNULL) {
+			@Override
+			public boolean tryAdvance(Consumer<? super ProteinSequence> action) {
+				if (closed) {
+					return false;
+				}
+				ProteinSequence protein = next(reader);
+				if (null == protein) {
+					return false;
+				}
+				action.accept(protein);
+				return true;
+			}
+
+			/**
+			 * Fetch the next header/protein tuple from the cache.  If the cache is empty, fetch another
+			 * batch from the source file
+			 *
+			 * @param reader
+			 * 		the input stream from which the FASTA content is read
+			 * @return the protein sequence
+			 */
+			private ProteinSequence next(FastaReader<ProteinSequence, AminoAcidCompound> reader) {
+				try {
+					if (!iterator.hasNext()) {
+						chunk = reader.process(getBatchSize());
+						if (null == chunk) {
+							closed = true;
+							reader.close();
+							return null;
+						}
+						iterator = chunk.entrySet().iterator();
+					}
+					if (iterator.hasNext()) {
+						Map.Entry<String, ProteinSequence> entry = iterator.next();
+						return createSequence(entry.getKey(), entry.getValue());
+					}
+					closed = true;
+					reader.close();
+				} catch (IOException exception) {
+					throw new RuntimeException(String.format("I/O error reading the FASTA file from '%s'", getPath()));
+				}
+				return null;
+			}
+		}; // Spliterator
+		return StreamSupport.stream(source, false);
+	}
+
+	/**
+	 * Create the sequence with the information from the header.  This implementation return the sequence as-is, but
+	 * this is an opportunity for the implementer to build specific information into the user collection space
+	 * of the sequence
+	 *
+	 * @param header the original header
+	 * @param sequence the protein sequence
+	 * @return the sequence
+	 */
+	protected ProteinSequence createSequence(String header, ProteinSequence sequence) {
```

Comment:
The `header` parameter is not used, is that intentional?

Reply:
Good catch.  The was in there before I found `getOriginalHeader`.